### PR TITLE
Photobox automation

### PR DESCRIPTION
**Pull request by waqasajaz**
Added support for login automation. Currently it takes hard coded credentials. work is in progress on pbx.coffee (starting file)

### CODE REVIEW

File: src/pbx.coffee [hunk 1]
```diff
@@ -0,0 +1,12 @@
+'use strict'
+account = require('./account')
+console.log 'Going to perform login'
+account.login('waqasajazch@gmail.com', '12345678').then ((status) ->
```
Comment (DinisCruz) [inline src/pbx.coffee]: this can be moved to an config.coffee file (loaded using ```require('../config') ```)

File: src/account.coffee [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+deferred = require('deferred')
+debug = require('debug')('login')
+websiteURL = 'https://www.photobox.co.uk/'
+scraperConf =
+  'timeout': 90000
+  'cookiesFile': '/tmp/photobox'
+  'loadImages': false
+  'ignoreSSLErrors': true
+  'webSecurity': false
+module.exports = login: (username, password) ->
```
Comment (DinisCruz) [inline src/account.coffee]: create CoffeeScript class and:
 - add login method to it
 - add scraperConf as properties of that class

File: src/account.coffee [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        else
+          debug 'Login unsuccessfull'
+          d.resolve false
+          horseman.close()
+    else
+      debug 'Already login.'
+      d.resolve true
+      return horseman.close()
+    return
+  d.promise
```
Comment (DinisCruz) [inline src/account.coffee]: this use of promises is confusing, we need to find a better and cleaner solution

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,15 @@
+FROM node:boron
+# Create app directory
```
Comment (DinisCruz) [inline Dockerfile]: Hi, why the use of boron? 

It is not the latest version right?

image

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+WORKDIR /usr/src/app
+# Install app dependencies
+COPY package.json package-lock.json ./
+# For npm@5 or later, copy package-lock.json as well
```
Comment (DinisCruz) [inline Dockerfile]: is it really needed the package-lock.json file in the docker version?

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+RUN npm install
+# If you are building your code for production
+# RUN npm install --only=production
+
+# Bundle app source
+COPY . .
+
+CMD [ "npm", "start" ]
```
Comment (DinisCruz) [inline Dockerfile]: 👍 nice touch (the docker file)

**Comment by DinisCruz** [comment]
Hi, in addition to the comments made above, can you move all files into a dedicated folder (at the moment these files are in the root folder of the repo

image

Also (for the next version, can you start adding tests?)

File: Dockerfile (hunk 5 above)
Comment (waqasajaz) [reply Dockerfile]: yes it is required. 

https://docs.npmjs.com/files/package-lock.json
"Describe a single representation of a dependency tree such that teammates, deployments, and continuous integration are guaranteed to install exactly the same dependencies."

https://nodejs.org/en/docs/guides/nodejs-docker-webapp/

**Pull request merged**